src/globus_sdk/services/search/data.py

```diff
@@ -221,6 +222,51 @@ def add_sort(
         return self
 
 
+class SearchQueryV1(utils.PayloadWrapper):
+    """
+    A specialized dict which has helpers for creating and modifying a Search
+    Query document. Replaces the usage of ``SearchQuery``.
+
+    :param q: The query string. Required unless filters are used.
+    :param limit: A limit on the number of results returned in a single page
+    :param offset: An offset into the set of all results for the query
+    :param advanced: Whether to enable (``True``) or not to enable (``False``) advanced
+        parsing of query strings. The default of ``False`` is robust and guarantees that
+        the query will not error with "bad query string" errors
+    :param additional_fields: additional data to include in the query document
+    """
+
+    def __init__(
+        self,
+        *,
+        q: str | utils.MissingType = utils.MISSING,
+        limit: int | utils.MissingType = utils.MISSING,
+        offset: int | utils.MissingType = utils.MISSING,
+        advanced: bool | utils.MissingType = utils.MISSING,
+        additional_fields: dict[str, t.Any] | utils.MissingType = utils.MISSING,
```

Comment:
Although it doesn't matter in terms of semantics, we conventionally put `additional_fields` and `query_params` (where each of these apply) at the end of the list of parameters. The reason being, they are the "escape valves" if none of the options above apply.
I'd like us to shift this one to match that style.